File: java-sdk/com/semantria/proxies/EntityUpdateProxy.java
package com.semantria.proxies;

import com.semantria.interfaces.IUpdateProxy;
import com.semantria.objects.configuration.UserEntity;

import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlElementWrapper;
import javax.xml.bind.annotation.XmlRootElement;
import java.util.ArrayList;
import java.util.List;

@XmlRootElement(name="entities")
public class EntityUpdateProxy implements IUpdateProxy<UserEntity>
{
	private List<UserEntity> added;
	private List<String> removed;
	
	public EntityUpdateProxy()
	{
		added = new ArrayList<UserEntity>();
		removed = new ArrayList<String>();
	}
	
	public void add(UserEntity item) 
	{
		added.add(item);
	}
	
	public void remove(UserEntity id)
	{
		removed.add(id.getName());
	}
	
	public void update(UserEntity item)
	{
		added.add(item);
	}

	public void clone(UserEntity item) {}

    public void removeAll() {
        added.clear();
        removed.clear();
        removed.add(REMOVE_ALL_ITEMS_MARK);
    }

	@XmlElementWrapper(name="added")
	@XmlElement(name="entity")
	public List<UserEntity> getAdded() { return added; }
	@XmlElementWrapper(name="removed")
	@XmlElement(name="entity")
	public List<String> getRemoved() { return removed; }
}

File: dotNet-sdk/Semantria/Proxy/EntityUpdateProxy.cs
using System;
using System.Collections.Generic;
using System.Text;
using Semantria.Com.Mapping.Configuration;
using Semantria.Com.Mapping.Configuration.Stub;

namespace Semantria.Com.Proxy
{
    public sealed class EntityUpdateProxy : IUpdateProxy<UserEntity>
    {
        #region Constructor

        private EntityUpdateProxy()
        {
        }

        internal static IUpdateProxy<UserEntity> CreateInstance()
        {
            return new EntityUpdateProxy();
        }

        #endregion Constructor

        #region Properties

        private static string REMOVE_ALL_ITEMS_MARK = "*";
        private EntityManagable _managable = new EntityManagable();
        internal EntityManagable Managable
        {
            get { return _managable; }
            set { _managable = value; }
        }

        #endregion Properties

        #region IUpdateProxy<Entity> Members

        object IUpdateProxy<UserEntity>.Stub
        {
            get { return _managable; }
        }

        void IUpdateProxy<UserEntity>.Add(UserEntity obj)
        {
            _managable.Added.Add(obj);
        }

        void IUpdateProxy<UserEntity>.Remove(UserEntity obj)
        {
            _managable.Removed.Add(obj.Name);
        }

        void IUpdateProxy<UserEntity>.Edit(UserEntity obj)
        {
            _managable.Added.Add(obj);
        }

        void IUpdateProxy<UserEntity>.Clone(UserEntity obj) {}

        void IUpdateProxy<UserEntity>.RemoveAll()
        {
            _managable.Added.Clear();
            _managable.Removed.Clear();
            _managable.Removed.Add(REMOVE_ALL_ITEMS_MARK);
        }

        #endregion
    }
}